sync/sync.go

```diff
@@ -617,3 +628,121 @@ func (t *Task) getToReplica(address string) (rest.Replica, error) {
 
 	return rest.Replica{}, fmt.Errorf("Failed to find target replica to copy to")
 }
+
+// InternalSnapshotCleaner should be run in the background, it tries to delete a snapshot every 60s
+// It fetches the checkpoint from controller which is present at in-memory of controller.
+// If checkpoint is not available at controller, snapshot delete is not initiated.
+// A deletion candidate list is prepared
+// In each iteration of the loop, one snapshot is picked from the top and deleted.
+func (t *Task) InternalSnapshotCleaner(s *replica.Server, repClient *replicaClient.ReplicaClient) {
+	ticker := time.NewTicker(SnapshotDeletionInterval)
+
+	for range ticker.C {
+		snapshot, err := t.client.GetCheckpoint()
+		if err == nil && snapshot == s.Replica().Info().Checkpoint {
```

Comment:
perform continue if err is not nil or snapshots are not equal
and how to handle if snapshots are not equal?

Reply:
added panic after 3 continuous mismatches